### ISSUE TITLE
kubernetes metrics logging needs to exclude irrelevant fstype's

### ISSUE THREAD

**Issue by eshicks4** [aws/amazon-cloudwatch-agent#635]
**Is your feature request related to a problem? Please describe.**
Our Cluster Insights currently contains 326,459 metrics under the "CWAgent > device, fstype, path" namespace.  It would appear that every NFS mount, overlay mount, etc. on every EKS node is being recorded.  Since containerd can create hundreds of mounts during normal operation, we need the vast majority of these ignored.

**Describe the solution you'd like**
I'm really only interested in the stats for the physical media as everything else is either irrelevant or recorded elsewhere.  Anything mounted via overlay, tmpfs, nfs4, etc. is not important to retain - especially many times each.

**Describe alternatives you've considered**
Support for the "disk" input type's "ignore_file_system_types" would be helpful but it would be better for everyone to just exclude certain fstypes by default.

**Additional context**
Cluster Insights is currently costing us over $3000 / month so this is rather important to us to address promptly.  Thanks

**Comment by eshicks4** [comment]
Here's a 1st shot at a list of fstype's to ignore:

`"ignore_file_system_types": [ "sysfs", "devtmpfs", "tmpfs", "overlay", "nfs", "nfs4" ]`

**Comment by venkivijay** [comment]
@wvh Did you set those values in ConfigMap? If so, can we add only the required metrics in ConfigMap? Please share a sample json blob.

**Comment by JerimyTate** [comment]
any progress on this?

**Comment by eshicks4** [comment]
The list above was just an example of what a solution might look like.  It isn't actually implemented.

FWIW, we couldn't sustain the cost any longer so I split out the FluentBit portion of Cluster Insights and routed it to our local Splunk instance.  We still lack operational metrics but FluentBit gets the logs we need for auditing and the rest just isn't worth the cost until this issue is resolved.

**Comment by venkivijay** [comment]
We moved away from CloudWatch Agent to [AWS Distro For OpenTelemetry Collector](https://github.com/aws-observability/aws-otel-collector). This way, we were able to filter out all the unnecessary metrics. We can also discard metrics based on k8 namespaces. Implementation is as easy a swapping out cloudwatch agent daemonset with ADOT collector and tweaking the pipeline yaml.  Hope this helps someone!
